### PR TITLE
Context - Custom errors

### PR DESCRIPTION
**Pull request by JuanMaGiraldo**
In this PR the sections of Context and Custom errors are covered.

### CODE REVIEW

File: lib/hello/accounts.ex [hunk 1]
```diff
@@ -0,0 +1,86 @@
+defmodule Hello.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+
+  import Ecto.Query, warn: false
```
Comment (juanhurtado10) [inline lib/hello/accounts.ex:6]: @JuanMaGiraldo Why is this import required?

File: lib/hello/accounts.ex [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  def list_users do
+    User
+    |> Repo.all()
+    |> Repo.preload(:credential)
+  end
+
+  def get_user!(id) do
+    User
+    |> Repo.get!(id)
+    |> Repo.preload(:credential)
```
Comment (juanhurtado10) [inline lib/hello/accounts.ex:21]: @JuanMaGiraldo what a preload does?

File: lib/hello/accounts.ex [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  alias Hello.Accounts.{User, Credential}
+
+
+  def list_users do
+    User
+    |> Repo.all()
+    |> Repo.preload(:credential)
+  end
+
+  def get_user!(id) do
```
Comment (juanhurtado10) [inline lib/hello/accounts.ex:18]: @JuanMaGiraldo when the bang `!` method is used?

File: lib/hello/accounts.ex [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  def update_user(%User{} = user, attrs) do
+    user
+    |> User.changeset(attrs)
+    |> Ecto.Changeset.cast_assoc(:credential, with: &Credential.changeset/2)
+    |> Repo.update()
+  end
+
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.changeset(attrs)
```
Comment (juanhurtado10) [inline lib/hello/accounts.ex:33]: @JuanMaGiraldo what a changeset does?

**Review by juanhurtado10 — APPROVED** [review]
@JuanMaGiraldo really good code and PR. Some Ecto questions were added

**Comment by juanhurtado10** [comment]
@JuanMaGiraldo also some conflicts need to be addressed.

File: lib/hello/accounts.ex (hunk 2 above)
Comment (JuanMaGiraldo) [reply lib/hello/accounts.ex:21]: `preload` preloads the associations, in this case, I'm preloading the schema Credential to use it with user, if I don't preload it, so, I'm just going to get the association as in the image:

image

File: lib/hello/accounts.ex (hunk 3 above)
Comment (JuanMaGiraldo) [reply lib/hello/accounts.ex:18]: It is a naming convention. The difference between a method with `!` and one that does not have it is that the bang method will raise an exception if the function encounters an error, and the other will only return the error.

File: lib/hello/accounts.ex (hunk 4 above)
Comment (JuanMaGiraldo) [reply lib/hello/accounts.ex:33]: With Changesets, we can create and modify models. It's a structure that stores the changes which are then passed to `Ecto.Repo` to be persisted. In addition, Changeset provides us validators that will be verified before persisting.

File: lib/hello/accounts.ex (hunk 1 above)
Comment (JuanMaGiraldo) [reply lib/hello/accounts.ex:6]: It's required to use the `Ecto.Query` functions like: from, select, where...
